File: src/PAhAom/GameState/Game.cpp
#include <PAhAom/GameState/GameState.hpp>
#include <PAhAom/Resources.hpp>
#include <PAhAom/Settings.hpp>

#include <util/Observer.hpp>

#include <iostream>

namespace PAhAom::GameState
{

Game::Game()
: currentLevel(dungeonGen)
{
        this->currentLevel.setPosition({8.f, 12.f});
}

void Game::handleInput(const sf::Event& event)
{
        if (event.type == sf::Event::KeyPressed)
        {
                switch (event.key.code)
                {
                        case sf::Keyboard::Escape:
                        {
                                util::Subject::send({ util::Message::PopState{} });
                                break;
                        }

                        default: break;
                }
        }
}

void Game::update(const util::DeltaTime dt)
{

}

void Game::draw(sf::RenderTarget& target)
{
        target.draw(this->currentLevel);
}

}

File: data/script/GameState/Game.lua
Game = {
    drawables = {
        {
            type = "sfText",
            content = "Game, press Esc to exit.",
            font = "unifont",
            position = {
                x = Settings.video.resolution.x / 2,
                y = Settings.video.resolution.y / 2,
            },
            fillColor = "red",
            origin = "center"
        }
    }
}

function Game:new(o)
    o = o or {}
    setmetatable(o, self)
    self.__index = self
    return o
end

function Game:handle_input(event)
    if event.type == "KeyPressed" then
        local kc = event.code
        if kc == Keyboard.Escape then
            pop_state()
        end
    end
end

function Game:update(dt)
end

function Game:draw()
    return self.drawables
end